File: web/src/core/feature/agent/domain/repository/agent.repository.ts
import type {
  AgentConnectionResult,
  AgentMessage,
  AgentReply,
} from "../entity/AgentMessage";

/**
 * Port — Agent gateway (Mistral). MCP tools llegan en Fase 2.
 */
export interface AgentRepository {
  /** Lightweight probe (models retrieve / list). */
  checkConnection(): Promise<AgentConnectionResult>;

  /**
   * Send a user turn to the configured Mistral Agent and return assistant text.
   * @param history optional prior messages (user/assistant only for Fase 1)
   */
  sendMessage(
    userText: string,
    history?: ReadonlyArray<Pick<AgentMessage, "role" | "content">>
  ): Promise<AgentReply>;
}

File: app/src/main/java/com/elitec/alejotaller/feature/agent/domain/repository/AgentRepository.kt
package com.elitec.alejotaller.feature.agent.domain.repository

import com.elitec.alejotaller.feature.agent.domain.entity.AgentConnectionResult
import com.elitec.alejotaller.feature.agent.domain.entity.AgentMessage
import com.elitec.alejotaller.feature.agent.domain.entity.AgentReply

interface AgentRepository {
    suspend fun checkConnection(): AgentConnectionResult

    suspend fun sendMessage(
        userText: String,
        history: List<AgentMessage> = emptyList(),
    ): AgentReply
}
